crates/core/app/tests/common/mod.rs

```diff
@@ -1,9 +1,32 @@
+//! Shared integration testing facilities.
+
+// NB: Allow dead code, these are in fact shared by files in `tests/`.
+#![allow(dead_code)]
+
 use async_trait::async_trait;
 use cnidarium::TempStorage;
 use penumbra_app::app::App;
 use penumbra_genesis::AppState;
 use std::ops::Deref;
 
+// Installs a tracing subscriber to log events until the returned guard is dropped.
+pub fn set_tracing_subscriber() -> tracing::subscriber::DefaultGuard {
+    use tracing_subscriber::filter::EnvFilter;
+
+    let filter = "debug,penumbra_app=trace,penumbra_mock_consensus=trace";
```

Comment:
this may need to have some custom statements in order to not make arkworks explode, see how the tracing filters are set in pcli for reference

Reply:
3c878f7c1

Reply:
for my future intuition, what does arkworks do with respect to our tests? i had seen that explicit disabling before, but figured that was due to pcli being part of the public interface with penumbra. does arkworks run our tests under some certain conditions?

Reply:
Really we should file a bug upstream.